packages/client/src/BrubeckNode.ts

```diff
@@ -170,12 +170,28 @@ export default class BrubeckNode implements Context {
     async openPublishProxyConnectionOnStreamPart(streamPartId: StreamPartID, nodeId: string): Promise<void> {
         try {
             if (!this.cachedNode || !this.startNodeComplete) {
-                const node = await this.startNode()
-                await node.joinStreamPartAsPurePublisher(streamPartId, nodeId)
+                await this.startNode()
             }
-            await this.cachedNode!.joinStreamPartAsPurePublisher(streamPartId, nodeId)
+            await Promise.all([
+                new Promise<void>((resolve, reject) => {
+                    this.cachedNode!.addPurePublishingAcceptedListener((node, stream) => {
+                        if (node === nodeId && stream === streamPartId) {
+                            resolve()
+                        }
+                    })
+                    this.cachedNode!.addPurePublishingRejectedListener((node, stream) => {
+                        if (node === nodeId && stream === streamPartId) {
+                            reject()
+                        }
+                    })
+                }),
+                this.cachedNode!.joinStreamPartAsPurePublisher(streamPartId, nodeId)
+            ])
+
         } finally {
             this.debug('openProxyConnectionOnStream << %o', streamPartId, nodeId)
+            this.cachedNode!.removePurePublishingAcceptedListener(() => {})
+            this.cachedNode!.removePurePublishingRejectedListener(() => {})
```

Comment:
I believe you need to pass in the same function references as you passed to the `on` listeners.

E.g.
```ts
const handlerFn = (node, stream) => {
    if (node === nodeId && stream === streamPartId) {
        resolve()
    }
}
this.cachedNode!.addPurePublishingAcceptedListener(handlerFn)
...
this.cachedNode!.removePurePublishingAcceptedListener(handlerFn)
```